reco_utils/recommender/deeprec/io/dkn_iterator.py

```diff
@@ -88,27 +106,22 @@ def parser_one_line(self, line):
         candidate_news_entity_index = []
         click_news_entity_index = []
 
-        for news in cols[1:]:
-            tokens = news.split(":")
-            if tokens[0] == "CandidateNews":
-                # word index start by 0
-                for item in tokens[1].split(","):
-                    candidate_news_index.append(int(item))
-                    candidate_news_val.append(float(1))
-            elif "clickedNews" in tokens[0]:
-                for item in tokens[1].split(","):
-                    click_news_index.append(int(item))
-                    click_news_val.append(float(1))
+        userid = cols[1]
+        candidate_news = cols[2]
 
-            elif tokens[0] == "entity":
-                for item in tokens[1].split(","):
-                    candidate_news_entity_index.append(int(item))
-            elif "entity" in tokens[0]:
-                for item in tokens[1].split(","):
-                    click_news_entity_index.append(int(item))
+        for item in self.news_word_index[candidate_news].split(","):
```

Comment:
We can use list comprehension, which is more efficient and clean.

Reply:
this case has two instructions:
```
        for item in self.news_word_index[candidate_news].split(","):
            candidate_news_index.append(int(item))
            candidate_news_val.append(float(1))
```
and there are several others. 

One can argue that showing it the way it is, it might be clearer.

I think both options are valid, so I would ask @Leavingseason if he wants to spend time in changing this

Reply:
Got it. Let's revisit this later if we'll have a performance issue.